WordPress/Classes/Networking/ServiceRemoteXMLRPC.m

```diff
@@ -0,0 +1,44 @@
+#import "ServiceRemoteXMLRPC.h"
+
+#import "WPXMLRPCClient.h"
+@interface ServiceRemoteXMLRPC()
+
+@property (nonatomic, strong, readwrite) WPXMLRPCClient *api;
+@property (nonatomic, copy) NSString *username;
+@property (nonatomic, copy) NSString *password;
+
+@end
+
+@implementation ServiceRemoteXMLRPC
+
+- (id)initWithApi:(WPXMLRPCClient *)api username:(NSString *)username password:(NSString *)password
+{
+    self = [super init];
+    if (self) {
+        _api = api;
+        _username = username;
+        _password = password;
+    }    
+    return self;
+}
+
+- (NSArray *)getXMLRPCArgsForBlogWithID:(NSNumber *)blogID extra:(id)extra
+{
+    NSMutableArray *result = [NSMutableArray array];
+    NSString *password = self.password ?: [NSString string];
+    NSString *username = self.username ?: [NSString string];
+    
+    [result addObject:blogID];
```

Comment:
So far, for self hosted sites we've been passing a `blogID=0`. I think it's mostly ignored server side, but I'd keep sending 0 to avoid any surprises. That means, no need to pass a blogID to this method, or to asset `blogID != nil` on the `RemoteXMLRPC` classes